hazelcast/src/test/java/com/hazelcast/client/executor/ClientExecutorServiceExceptionTest.java

```diff
@@ -0,0 +1,97 @@
+import com.hazelcast.cluster.Member;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IExecutorService;
+import com.hazelcast.spi.exception.RetryableIOException;
+import com.hazelcast.spi.exception.TargetNotMemberException;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.Serializable;
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class ClientExecutorServiceExceptionTest {
+
+    private final TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
+
+    @After
+    public void tearDown() {
+        hazelcastFactory.terminateAll();
+    }
+
+
+    @Test(expected = TargetNotMemberException.class)
+    public void testSubmitToNonMember() throws Throwable {
+        hazelcastFactory.newHazelcastInstance();
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getNetworkConfig().setSmartRouting(false);
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient(clientConfig);
+        IExecutorService executorService = client.getExecutorService("test");
+
+        HazelcastInstance instance2 = hazelcastFactory.newHazelcastInstance();
+        Member member2 = instance2.getCluster().getLocalMember();
+
+        instance2.shutdown();
+
+        try {
+            executorService.submitToMember((Serializable & Callable<String>) () -> "test", member2).get();
+        } catch (Exception e) {
+            throw e.getCause();
+        }
+    }
+
+    public static class SecondTimeSuccessCallable implements Serializable, Callable {
+        private static AtomicInteger runCount = new AtomicInteger();
+
+
+        @Override
+        public Object call() throws Exception {
+            if (runCount.incrementAndGet() == 1) {
+                throw new RetryableIOException();
+            }
+            return "SUCCESS";
+        }
+    }
+
+    @Test
+    public void testRetriableIOException() throws Throwable {
+        hazelcastFactory.newHazelcastInstance();
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getNetworkConfig().setSmartRouting(false);
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient(clientConfig);
+        IExecutorService executorService = client.getExecutorService("test");
+
+        HazelcastInstance instance2 = hazelcastFactory.newHazelcastInstance();
+        Member member2 = instance2.getCluster().getLocalMember();
+        assertEquals("SUCCESS", executorService.submitToMember(new SecondTimeSuccessCallable(), member2).get());
```

Comment:
did you want to shutdown member1 in this test?

Reply:
No, I don't but I understand the confusion. Let me rewrite the test by using a single member. The second member is actually not needed for this test.